crates/turborepo-lib/src/run/summary/task.rs

```diff
@@ -74,3 +94,94 @@ impl<'a> TaskSummary<'a> {
         self.package = None;
     }
 }
+
+impl TaskCacheSummary {
+    pub fn cache_miss() -> Self {
+        Self {
+            local: false,
+            remote: false,
+            status: CacheStatus::Miss,
+            time_saved: 0,
+            source: None,
+        }
+    }
+}
+
+impl From<Option<CacheResponse>> for TaskCacheSummary {
+    fn from(value: Option<CacheResponse>) -> Self {
+        value.map_or_else(Self::cache_miss, |CacheResponse { source, time_saved }| {
+            let source = CacheSource::from(source);
+            // Assign these deprecated fields Local and Remote based on the information
+            // available in the itemStatus. Note that these fields are
+            // problematic, because an ItemStatus isn't always the composite
+            // of both local and remote caches. That means that an ItemStatus might say it
+            // was a local cache hit, and we return remote: false here. That's misleading
+            // because it does not mean that there is no remote cache hit,
+            // it _could_ mean that we never checked the remote cache. These
+            // fields are being deprecated for this reason.
+            let (local, remote) = match source {
+                CacheSource::Local => (true, false),
+                CacheSource::Remote => (false, true),
+            };
+            Self {
+                local,
+                remote,
+                status: CacheStatus::Hit,
+                source: Some(source),
+                time_saved,
+            }
+        })
+    }
+}
+
+impl From<turborepo_cache::CacheSource> for CacheSource {
+    fn from(value: turborepo_cache::CacheSource) -> Self {
+        match value {
+            turborepo_cache::CacheSource::Local => Self::Local,
+            turborepo_cache::CacheSource::Remote => Self::Remote,
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use serde_json::json;
+    use test_case::test_case;
+
+    use super::*;
+
+    #[test_case(CacheStatus::Hit, json!("HIT") ; "hit")]
+    #[test_case(CacheStatus::Miss, json!("MISS") ; "miss")]
+    #[test_case(CacheSource::Local, json!("LOCAL") ; "local")]
+    #[test_case(CacheSource::Remote, json!("REMOTE") ; "remote")]
+    #[test_case(
+        TaskCacheSummary::cache_miss(),
+        serde_json::json!({
+                "local": false,
+                "remote": false,
+                "status": "MISS",
+                "timeSaved": 0,
+            })
+        ; "cache miss"
+    )]
+    #[test_case(
+        TaskCacheSummary {
+            local: true,
+            remote: false,
+            status: CacheStatus::Hit,
+            source: Some(CacheSource::Local),
+            time_saved: 6,
+        },
+        serde_json::json!({
```

Comment:
This is the expected schema as shown in [run-summary/monorepo.t](https://github.com/vercel/turbo/blob/main/turborepo-tests/integration/tests/run-summary/monorepo.t#L106)